c-sharp-razor/src/Minitwit.Infrastructure/Repository/AuthorRepository.cs

```diff
@@ -0,0 +1,269 @@
+using Microsoft.EntityFrameworkCore;
+using Minitwit.Core.Entities;
+using Minitwit.Core.Repository;
+
+namespace Minitwit.Infrastructure.Repository;
+
+public class AuthorRepository : BaseRepository, IAuthorRepository
+{
+    private readonly IFollowRepository _followRepository;
+
+    public AuthorRepository(MinitwitDbContext minitwitDbContext, IFollowRepository followRepository)
+        : base(minitwitDbContext)
+    {
+        _followRepository = followRepository;
+    }
+
+    // ----- Add Author Methods ----- //
+    public async void AddAuthorAsync(Author author)
+    {
+        await db.Users.AddAsync(author);
+        await db.SaveChangesAsync();
+    }
+
+    // ----- Get Author Methods ----- //
+
+    public async Task<ICollection<Author>> GetAllAuthorsAsync()
+    {
+        return await db.Users.ToListAsync();
+    }
+
+    public async Task<ICollection<Author>> GetAuthorsByIdAsync(IEnumerable<int> authorIds)
+    {
+        return await db.Users.Where(a => authorIds.Contains(a.Id)).AsNoTracking().ToListAsync();
+    }
+
+    public async Task<Author?> GetAuthorByIdAsync(int authorId)
+    {
+        Author? author = await db.Users.FirstOrDefaultAsync(a => a.Id == authorId);
+        return author!;
+    }
+
+    public async Task<Author> GetAuthorByNameAsync(string name)
+    {
+        Author? author = await db.Users.FirstOrDefaultAsync(a => a.UserName == name)!;
+        return author!;
+    }
+
+    public async Task<Author> GetAuthorByEmail(string email)
+    {
+        Author? author = await db.Users.FirstOrDefaultAsync(a => a.Email == email)!;
+        return author!;
+    }
+
+    // ----- Get Messages By Author and Page Methods ----- //
+    public async Task<ICollection<Message>> GetMessagesByAuthorAsync(int id)
+    {
+        return await db.Messages.Where(e => e.AuthorId == id).ToListAsync();
+    }
+
+    public async Task<ICollection<Message>> GetMessagesByAuthor(int id, int page)
+    {
+        var Messages = await GetMessagesByAuthorAsync(id);
+
+        //Check that author has Messages
+        if (Messages == null || Messages.Count == 0)
+            throw new Exception("This author has no Messages");
+
+        if (page < 1)
+            page = 1;
+
+        int pageSizeIndex = (page - 1) * PageSize;
+
+        if (Messages.Count < pageSizeIndex + PageSize)
+            return Messages
+                .ToList()
+                .GetRange(pageSizeIndex, Messages.Count - pageSizeIndex)
+                .OrderByDescending(c => c.TimeStamp)
+                .ToList();
+        if (Messages.Count > PageSize)
+            return Messages
+                .ToList()
+                .GetRange(pageSizeIndex, PageSize)
+                .OrderByDescending(c => c.TimeStamp)
+                .ToList();
+        return Messages.OrderByDescending(c => c.TimeStamp).ToList();
+    }
+
+    public async Task<ICollection<Message>> GetMessagesByAuthorAndFollowingAsync(int id)
```

Comment:
What is the difference between this method and the next one? Both are being used?